core/contracts/financial_templates/PricelessPositionManager.sol

```diff
@@ -453,6 +453,30 @@ contract PricelessPositionManager is FeePayer {
         oracle.requestPrice(priceIdentifer, requestedTime);
     }
 
+    function _payOracleRequestFees() private {
+        // Send the fee payment.
+        FixedPoint.Unsigned memory totalPaid = _payFinalFees(address(this));
+
+        // Exit early if totalCollaterall == 0 to prevent divide by 0.
+        if (totalPositionCollateral.isEqual(FixedPoint.fromUnscaledUint(0))) {
```

Comment:
Instead of checking whether the totalPositionCollateral == 0, shouldn't we make sure that it's greater than the fees (and revert if it's not)? Otherwise, there's not enough collateral in the live positions to pay for the final fee, so the collateral is effectively being pulled from the liquidation pool, meaning someone may not be able to withdraw their liquidation reward.

Reply:
yes I think so, but shouldn't we do the same for `payFees`?

```
// Exit early if pfc == 0 to prevent divide by 0.
        if (initialPfc.isEqual(FixedPoint.fromUnscaledUint(0))) {
            return totalPaid;
        }
```

Reply:
As discussed in IRL, since final fee is a flat amount and not a %, its possible for `finalFee >= totalPositionCollateral` which will either revert `expire` or render it useless. I replaced the `if` statement above with `require(totalPositionCollateral.isGreaterThan(totalPaid))`

This does raise an edge case in which collateral can be locked in the contract, which I've brought up in #925